src/components/views/dialogs/MultiInviteDialog.js

```diff
@@ -0,0 +1,199 @@
+            if (getAddressType(input) === null) {
+                this.state.completionStates[i] = 'error';
+                this.state.errorTexts[i] = 'Unrecognised address';
+            }
+        }
+    }
+
+    _onCancel() {
+        this.props.onFinished(false);
+    }
+
+    _startInviting() {
+        this.setState({
+            completionStates: [],
+            busy: true,
+            done: false,
+        });
+        this._inviteMore(0);
+    }
+
+    _inviteMore(nextIndex) {
+        if (nextIndex == this.props.inputs.length) {
+            this.setState({
+                busy: false,
+                done: true,
+            });
+            return;
+        }
+
+        const input = this.props.inputs[nextIndex];
+
+        // don't try to invite it if it's an invalid address
+        // (it will already be marked as an error though,
+        // so no need top do so again
+        if (getAddressType(input) === null) {
+            this._inviteMore(nextIndex + 1);
+            return;
+        }
+
+        // don't re-invite (there's no way in the UI to do this, but
+        // for sanity's sake)
+        if (this.state.completionStates[nextIndex] == 'invited') {
+            this._inviteMore(nextIndex + 1);
+            return;
+        }
+
+        inviteToRoom(this.props.roomId, input).then(() => {
+            this.setState((s) => {
+                s.completionStates[nextIndex] = 'invited'
+                return s;
+            });
+            this._inviteMore(nextIndex + 1);
+        }, (err) => {
+            let errorText;
+            let fatal = false;
+            if (err.errcode == 'M_FORBIDDEN') {
+                fatal = true;
+                errorText = 'You do not have permission to invite people to this room.';
+            } else if (err.errcode == 'M_LIMIT_EXCEEDED') {
+                // we're being throttled so wait a bit & try again
+                setTimeout(() => {
+                    this._inviteMore(nextIndex);
+                }, 5000);
+                return;
+            } else {
+                errorText = 'Unknown server error';
+            }
+            this.setState((s) => {
+                s.completionStates[nextIndex] = 'error';
+                s.errorTexts[nextIndex] = errorText;
+                s.busy = !fatal;
+                s.done = fatal;
+                return s;
+            });
+            if (!fatal) {
+                this._inviteMore(nextIndex + 1);
+            }
+        });
+    }
+
+    _getProgressIndicator() {
+        const numErrors = this.state.completionStates.filter((s) => {
+            return s == 'error';
+        }).length;
+        let errorText;
+        if (numErrors > 0) {
+            const plural = numErrors > 1 ? 's' : '';
+            errorText = <span className="error">({numErrors} error{plural})</span>
+        }
+        return <span>
+            {this.state.completionStates.length} / {this.props.inputs.length} {errorText}
```

Comment:
if you have 10 inputs, of which the last is an invalid address, and you haven't started doing any invites yet, `completionStates.length` will be 10; I don't think that's what you want.